golem-rib/src/interpreter/rib_runtime_error.rs

```diff
@@ -0,0 +1,234 @@
+use crate::interpreter::interpreter_stack_value::RibInterpreterStackValue;
+use crate::{InstructionId, TypeHint};
+use golem_wasm_rpc::{Value, ValueAndType};
+use std::fmt::{Display, Formatter};
+
+#[derive(Debug)]
+pub enum RibRuntimeError {
+    ArithmeticError {
+        message: String,
+    },
+    CastError {
+        from: CastFrom,
+        to: TypeHint,
+    },
+    ExhaustedIterator,
+    FieldNotFound {
+        field: String,
+    },
+    FunctionInvokeError {
+        function_name: String,
+        error: Box<dyn std::error::Error + Send + Sync>,
+    },
+    IndexOutOfBound {
+        index: usize,
+        size: usize,
+    },
+    InfiniteComputation {
+        message: String,
+    },
+    InputNotFound(String),
+    InvariantViolation(InvariantViolation),
```

Comment:
Hopefully in future, we can confidently unwrap for violation of the the invariant.
Also in future, we can add or remove more errors into this enum - to get a clear idea of which one can be user exposed or not.

For now, its just better than what was before